styleguide/STYLEGUIDE.md

```diff
@@ -1,50 +1,1090 @@
+```
+
+**NOTE**: Swift is very different from JavaScript, where omitting semicolons is [generally considered unsafe](http://stackoverflow.com/questions/444080/do-you-recommend-using-semicolons-after-every-statement-in-javascript)
+
+## Parentheses
+
+Parentheses around conditionals are not required and should be omitted.
+
+**Preferred**:
+```swift
+if name == "Hello" {
+  print("World")
+}
+```
+
+**Not Preferred**:
+```swift
+if (name == "Hello") {
+  print("World")
+}
+```
+
+In larger expressions, optional parentheses can sometimes make code read more clearly.
+
+**Preferred**:
+```swift
+let playerMark = (player == current ? "X" : "O")
+```
+
+## Multi-line String Literals
+
+When building a long string literal, you're encouraged to use the multi-line string literal syntax. Open the literal on the same line as the assignment but do not include text on that line. Indent the text block one additional level.
+
+**Preferred**:
+
+```swift
+let message = """
+  You cannot charge the flux \
+  capacitor with a 9V battery.
+  You must use a super-charger \
+  which costs 10 credits. You currently \
+  have \(credits) credits available.
+  """
+```
+
+**Not Preferred**:
+
+```swift
+let message = """You cannot charge the flux \
+  capacitor with a 9V battery.
+  You must use a super-charger \
+  which costs 10 credits. You currently \
+  have \(credits) credits available.
+  """
+```
+
+**Not Preferred**:
+
+```swift
+let message = "You cannot charge the flux " +
+  "capacitor with a 9V battery.\n" +
+  "You must use a super-charger " +
+  "which costs 10 credits. You currently " +
+  "have \(credits) credits available."
+```
+
+## No Emoji
+
+Do not use emoji in your projects. For those readers who actually type in their code, it's an unnecessary source of friction. While it may be cute, it doesn't add to the learning and it interrupts the coding flow for these readers.
+
+## No #imageLiteral or #colorLiteral
+
+Likewise, do not use Xcode's ability to drag a color or an image into a source statement. These turn into #colorLiteral and #imageLiteral, respectively, and present unpleasant challenges for a reader trying to enter them based on tutorial text. Instead, use `UIColor(red:green:blue)` and `UIImage(imageLiteralResourceName:)`.
+
+## Assertions and Fatal Errors
 
 We use fatal errors when the app reaches a state that shouldn't happen in a release build and isn't handled, for example, if we fail to instantiate a view controller from a storyboard.
 
 We use assert() and assertionFailure() also for when something happens that shouldn't, but when the resulting app state is explicitly handled and is recoverable.
 
-### Unit test names
+## Logging
+
+We use [unified logging system](https://developer.apple.com/documentation/os/logging) to capture log messages. To send a message to the logging system, call the `os_log` function. Please use debug log level to protect users' privacy.
+
+## Unit Test Names
 
 * We use the when then convention for test:
 
```

Comment:
LGTM! Small comment from me: overall thanks for putting it all together, nice work, really! I would like to spend the next quick wins days by introducing SwiftFormat and perhaps somehow merging this list of yours with Airbnb code style guide https://github.com/airbnb/swift so it could benefit in two ways: we would have rules in SwiftFormat defined and mapped to rules in the guide so whenever someone's developing smth, the code could be "fixed" with a single shortcut or pre-commit hook and 2nd during the PRs in comments we could quickly link to certain rules inside the style guide, like this: https://github.com/airbnb/swift#use-anyobject Please let me know what you think ;)